arch/arm/core/cortex_a_r/reset.S

```diff
@@ -203,12 +203,14 @@ EL1_Reset_Handler:
 #if CONFIG_MP_MAX_NUM_CPUS > 1
     get_cpu_id r1
 
+2:
     ldrex r2, [r0, #BOOT_PARAM_MPID_OFFSET]
     cmp r2, #-1
     bne 1f
     strex r3, r1, [r0, #BOOT_PARAM_MPID_OFFSET]
     cmp r3, #0
     beq _primary_core
+    b 2b
```

Comment:
Does MMU disablement cause the exclusive monitor issue? If so, this should be fixed as this one https://github.com/zephyrproject-rtos/zephyr/pull/63905

Reply:
I have observed this with cacheability enabled (on AArch32-R you can get cacheable memory without specifying any memory ranges in the MPU, as PMSAv8-32 has a "background" default memory map with a cacheable range).

However, using an explicit logic ID like your AArch64 fix would also work for this configuration, and avoids relying on the exclusive monitor altogether.

Reply:
Yes, When MMU is disabled, using these Load-Exclusives instructions may have the same issues on aarch32. So, using Voting lock algorithm is necessary.